torchtext/datasets/dbpedia.py

```diff
@@ -28,13 +28,23 @@
 
 @_add_docstring_header(num_lines=NUM_LINES, num_classes=14)
 @_create_dataset_directory(dataset_name=DATASET_NAME)
-@_wrap_split_argument(('train', 'test'))
-def DBpedia(root, split):
-    dataset_tar = download_from_url(URL, root=root,
-                                    path=os.path.join(root, _PATH),
-                                    hash_value=MD5, hash_type='md5')
-    extracted_files = extract_archive(dataset_tar)
-
-    path = _find_match(split + '.csv', extracted_files)
-    return _RawTextIterableDataset(DATASET_NAME, NUM_LINES[split],
-                                   _create_data_from_csv(path))
+@_wrap_split_argument(("train", "test"))
+def DBpedia(root: str, split: Union[Tuple[str], str]):
+    # TODO Remove this after removing conditional dependency
+    if not is_module_available("torchdata"):
+        raise ModuleNotFoundError("Package `torchdata` not found. Please install following instructions at `https://github.com/pytorch/data`")
+
+    url_dp = IterableWrapper([URL])
+
+    cache_dp = url_dp.on_disk_cache(
+        filepath_fn=lambda x: os.path.join(root, _PATH),
+        hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
+    )
+    cache_dp = GDriveReader(cache_dp).end_caching(mode="wb", same_filepath_fn=True)
+    cache_dp = FileOpener(cache_dp, mode="b")
+
+    extracted_files = cache_dp.read_from_tar()
+
+    filter_extracted_files = extracted_files.filter(lambda x: split + ".csv" in x[0])
```

Comment:
Same comment as in the other PR. Let's make this consistent with the logic [here](https://github.com/pytorch/text/pull/1503/files#diff-8342c3e1b0563ee8e6983216c4660087d8c94d6b641e668780cac56c15227388R53)

Reply:
Note that the "expected files" map doesn't really exist here. I can make it exist -- should I do that in this PR or a separate one?

Reply:
I think it should be fine to do that in this PR. 

Let's verify with @parmeet first to see if there's reason that `_EXTRACTED_FILES` and `_EXTRACTED_FILES_MD5` weren't created for this dataset.